java/arcs/core/storage/ReferenceModeStore.kt

```diff
@@ -479,6 +480,18 @@ class ReferenceModeStore private constructor(
         return backingStore.onProxyMessage(ProxyMessage.Operations(op, id = null), referencable.id)
     }
 
+    /** Clear all entities from the backing store, using the container store to retrieve the ids. */
+    private suspend fun clearAllEntitiesInBackingStore(): Boolean {
+        val containerModel = containerStore.getLocalData()
+        if (containerModel !is CrdtSet.Data<*>) {
+            throw UnsupportedOperationException()
+        }
+        return containerModel.values.all { (refId, data) ->
+            val clearOp = listOf(CrdtEntity.Operation.ClearAll(crdtKey, data.versionMap))
```

Comment:
note: data.versionMap is the version at which the entity was added to the set. Here you are passing it as the entity-modification version map, they are different versions with different actors (the entity-modification versions are generated in this file).
maybe you can use versions.getOrPut(entity.id)

Reply:
For future reference: this should be ok, since this function is clearing the backing entities (by zeroing them out as per `clearEntityInBackingStore`), so using the entity-mod version map is appropriate.

Reply:
Yes we would like to use the entity-mod version map. But here you are using the ref-added-to-set versionmap. See the comment here http://google3/third_party/java_src/arcs/java/arcs/core/crdt/CrdtSet.kt?l=181&rcl=319892547.
The entity-mod version map is inside the reference.

maybe you can use if (containerModel !is CrdtSet.Data<Reference>) { ...
then 
ClearAll(crdtKey, data.value.version)